src/rai_bench/rai_bench/tool_calling_agent/mocked_tools.py

```
@@ -265,45 +268,163 @@ def _run(self, msg_type: str) -> str:
+            validation fails.
+        """
+        if service_type not in self.custom_models:
+            raise ValueError(f"Service type: {service_type} is invalid custom type")
+
+        model = self.custom_models[service_type]
+        try:
+            model.model_validate(args)
+        except ValidationError as e:
+            raise ValueError(f"Pydantic validation failed: {e}")
+
+    def validate(self, service_type: str, args: Dict[str, Any]):
```

Pylint (3.3.7)

[refactor] Either all return statements in a function should return an expression, or none of them should.

(R1710)